lib/service/tezos_beacon_service.dart

```diff
@@ -71,6 +72,7 @@ class TezosBeaconService implements BeaconHandler {
         await Future.delayed(const Duration(seconds: 1));
       }
     } while (retryCount < maxRetries);
+    if (retryCount > maxRetries) memoryValues.deepLink.value = null;
```

Comment:
this will never execute. `retryCount` will never greater than `maxRetries`. It will stop when they are equal.